src/librustc_mir/transform/const_prop.rs

```diff
@@ -259,19 +259,15 @@ impl<'mir, 'tcx> ConstPropagator<'mir, 'tcx> {
                 use rustc::mir::interpret::InterpError::*;
                 match diagnostic.error {
                     Exit(_) => bug!("the CTFE program cannot exit"),
-                    | Unsupported(_) => {},
-                    | UndefinedBehaviour(_) => {},
-                    | InvalidProgram(_) => {},
-                    | ResourceExhaustion(_) => {},
-                    | Panic(_)
-                    => {
+                    Panic(_) => {
                         diagnostic.report_as_lint(
                             self.ecx.tcx,
                             "this expression will panic at runtime",
                             lint_root,
                             None,
                         );
                     }
+                    _ => {},
```

Comment:
I feel uneasy about the catch-all here, after all this was an exhaustive `match` before. Please keep it exhaustive.

What I meant was that you can do something like
```
Unsupported(_) | UndefinedBehaviour(_) | ... => {
  // Ignore these errors.
}
```